.github/workflows/rust_ci.yml

```diff
@@ -0,0 +1,87 @@
+# This CI (Continuous Integration) file will automatically check your Rust code for errors.
+# It runs using GitHub Actions: https://docs.github.com/en/actions
+# It will check the code compiles, run tests, run lints, and check for security issues.
+# CI will help you standardise your code style and to detect issues with your code easily and early.
+# It makes it easier to integrate different branches once they're finished.
+# adapted from https://github.com/actions-rs/meta/blob/master/recipes/quickstart.md and https://gist.github.com/LukeMathWalker/5ae1107432ce283310c3e601fac915f3
+
+name: Rust CI
+
+on:
+  push:
+  #     branches:
+  #       - main
+  release:
+    types: [published]
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  check:
+    name: Check code compiles
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+      - name: Run cargo check
+        run: cargo check
+
+  test:
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
```

Comment:
Two things with this:
- LogicRs requires the nightly toolchain (idk if that's available with this toolchain provider)
- I'd prefer if we used the official rust solution and not depend on third-party tools. I'm not _that_ into CI, so if there's a good reason against that, please lmk

Reply:
Hey, thanks for your feedback.

To your first point, yeah, I'll change it over to the nightly toolchain if that's what your code needs.

To your second, [there isn't an official Rust GitHub action, but dtolnay/rust-toolchain is now what most people use](https://github.com/rust-lang/rustup/issues/3409). I think the checkout action has been updated though, so I'll update it and also recheck everything else is up-to-date too.

Reply:
OK @Spydr06,

I just finished moving all the actions over to the nightly toolchain, updating the checkout action used, and moving everything over to the de-facto standard actions used for jobs. Please let me know if you have any further thoughts. Thanks! :)

Reply:
> To your second, https://github.com/rust-lang/rustup/issues/3409. I think the checkout action has been updated though, so I'll update it and also recheck everything else is up-to-date too.

Ah ok, did not know that. Then it's fine ofc